Updates EME demo to support Widevine and unprefixed EME APIs (#120)
MicrosoftEdge/Demos: .github/copilot-instructions.md

# Copilot Coding Agent Instructions

## Repository Overview

**MicrosoftEdge/Demos** is a collection of standalone web demo applications and sample code used to demonstrate Microsoft Edge browser features (DevTools, PWAs, cross-browser APIs, extensions). Each demo lives in its own top-level directory and is independently deployable.

Demos are hosted as GitHub Pages at `https://microsoftedge.github.io/Demos/<demo-name>/`.

## Repository Structure

```
/                         <- repo root
├── .github/              <- GitHub config (this file lives here)

> ├── .hintrc               <- webhint linting configuration (browserslist)

Reviewer: Remove this, we actually don't really use the .hintrc file.

├── .gitignore
├── README.md             <- index of all demos, grouped by category
├── template/             <- minimal template to copy when adding a new demo
│   ├── index.html
│   ├── style.css
│   ├── script.js

> │   └── README.md

Reviewer: Remove this, we actually don't copy from the template anymore.

├── <demo-name>/          <- each demo is self-contained in its own directory
│   ├── index.html
│   ├── README.md         <- required; includes live demo link
│   └── ...
└── ...
```

### Demos with a build step

Most demos are pure HTML/CSS/JS with no build step. A few have a `package.json` and use tools like webpack or TypeScript:

| Directory | Build tool | Build command |
|---|---|---|
| `/1DIV/` | webpack | `npm install && npx webpack` |
| `/slow-calendar/` | webpack + TypeScript | `npm install && npm run build` |
| `/devtools-crash-analyzer/` | webpack + TypeScript | `npm install && npm run build-prod` |
| `/built-in-ai/` | Eleventy (SSG) | `npm install && npm run build` |
| `/devtools-websocket-demo/` | Node.js server | `npm install && npm run server` |
| `/heap-snapshot-visualizer/` | webpack | `npm install && npm run build` (check local `package.json`) |
| `/css-mirroring-sourcemaps-demo/` | (check local `package.json`) | `npm install && npm run build` |

Always check the directory's own `package.json` for available scripts before building.

## Adding a New Demo

> 1. Copy `/template/` to a new top-level directory (e.g., `/my-demo/`).

Reviewer: ```suggestion
1. Create a new top-level directory (e.g., `/my-demo/`).
```

> 2. Edit `README.md` in the new directory:

Reviewer: ```suggestion
2. Create `README.md` in the new directory:
```

   - Describe what the demo does.

>    - Include a live link: `https://microsoftedge.github.io/Demos/my-demo/`.

Reviewer: ```suggestion
   - Add a self-explanatory title as a h1 header.
   - Immediately below the title, add a link to the live demo: `➡️ **[Open the demo](https://microsoftedge.github.io/Demos/my-demo/)** ⬅️`
   - Add a paragraph that describes what the demonstrated feature is about, and what the demo does. Provide any links to specs or docs about the demonstrated feature.
   - Include instructions to enable the feature in Edge if needed.
```

> 3. Edit `index.html`, `style.css`, `script.js` as needed.

Reviewer: ```suggestion
3. Create an `index.html` file for the demo. Prefer embedded all necessary CSS and JS code inside the file directly, but extract to `style.css` and `script.js` files as needed.
```

4. Add a row to the appropriate table in the root `README.md`.

## Coding Conventions

- **No global build system.** There is no root-level `package.json`, `Makefile`, or test runner. Each demo is independent.
- **No test infrastructure.** There are no automated tests anywhere in the repo. Do not add or run tests.
- **Minimal dependencies.** Demos are plain HTML/CSS/JS whenever possible. Only add npm packages if the demo genuinely needs them.
- **Self-contained directories.** Each demo directory contains everything it needs. Do not create shared utility packages across demos.
- **Node modules excluded from git.** The `.gitignore` excludes `node_modules/` globally. Never commit `node_modules/`.

> - **Browser compatibility.** `.hintrc` configures webhint with a `browserslist` targeting modern browsers (defaults, no IE 11, no Firefox ≤ 91, no iOS Safari ≤ 14.8).

Reviewer: ```suggestion
```

## Linting

The repo uses [webhint](https://webhint.io/) configured via `.hintrc`. There is no npm script at the root to run it. If you need to lint, install and run webhint per-demo:
```bash
cd <demo-directory>
npx hint .

> ```

Reviewer: Remove this section.
